### PR TITLE
Add a choose letter type step to LA Letter Builder.

### PR DESCRIPTION
**Pull request by samaratrilling**
Screen Shot 2021-12-10 at 12 44 53 PM

### CODE REVIEW

**Comment by samaratrilling** [comment]
Made the question required. I haven't figured out how to get any previously-saved values for the letter type to show up (that requires writing a resolver for the session info thing, I think, but I couldn't get the types to work). That'll be next.

File: laletterbuilder/schema.py [hunk 1]
```diff
@@ -25,3 +27,17 @@ def update_onboarding_info(cls, form, info: Dict[str, Any]):
...
+    class Meta:
+        form_class = forms.ChooseLetterForm
+
+    login_required = False
+
+    @classmethod
+    def perform_mutate(cls, form, info: ResolveInfo):
+        # TODO: have this set a variable that decides which
+        # path the user goes down - perhaps just a redirect
+        return cls.mutation_success()
```
Comment (sraby) [inline laletterbuilder/schema.py:43]: why is the decided path something that needs to be saved server-side? Couldn't it just be a front-end link?

**Review by sraby — CHANGES_REQUESTED** [review]
Looking great @samaratrilling! The route structure looks good as well as the tsx component. I'm still curious what the role of `LaLetterBuilderChooseLetterMutation` is here given that we aren't committing the letter type choice to the server at all? My thinking was that the page where the user selects what letter is simply a list of links that direct users to the right route? Maybe there's an added element that I'm missing here

File: frontend/lib/laletterbuilder/letter-builder/choose-letters.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+} from "../../../../common-data/la-letter-builder-letter-choices";
+import { SessionUpdatingFormSubmitter } from "../../forms/session-updating-form-submitter";
+import Page from "../../ui/page";
+import { MiddleProgressStep } from "../../progress/progress-step-route";
+import { LaLetterBuilderChooseLetterMutation } from "../../queries/LaLetterBuilderChooseLetterMutation";
+
+export const LaLetterBuilderChooseLetterStep = MiddleProgressStep((props) => {
+  return (
+    <Page title={li18n._(t`Letter type you'd like to send`)}>
+      <SessionUpdatingFormSubmitter
```
Comment (sraby) [inline frontend/lib/laletterbuilder/letter-builder/choose-letters.tsx:21]: If this page is simply navigating a user to the right path (depending on letter type), I'm wondering if it even needs to be a Session-updating form at all? Couldn't it be just a list of links? Or, if you wanted to keep the radio buttons, you could just save the radio button selection on the front end, and that dynamically determines what the "next" button goes to?

File: frontend/lib/analytics/amplitude.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import JustfixRoutes from "../justfix-route-info";
 import { NorentRoutes } from "../norent/route-info";
 import { EvictionFreeRoutes } from "../evictionfree/route-info";
 import { USER_ID_PREFIX } from "../../../common-data/amplitude";
-import { LaLetterBuilderRoutes } from "../laletterbuilder/route-info";
+import { LaLetterBuilderRouteInfo } from "../laletterbuilder/route-info";
```
Comment (samaratrilling) [inline frontend/lib/analytics/amplitude.ts:17]: It was confusing me to no end that the stuff from route-info.tsx was called Routes and the stuff from routes.tsx was called something else. I changed it to make it make more sense.

Comment (sraby) [reply frontend/lib/analytics/amplitude.ts:17]: thank you!!

**Review by sraby — APPROVED** [review]
Nice!!

**Pull request merged**